node/src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -160,6 +163,14 @@ delTxExtraWithHistory id addrs = do
     for_ addrs $ modifyAddrHistory $
         NewestFirst . delete id . getNewestFirst
 
+updateUtxoSumFromBalanceUpdate :: MonadTxExtra m => BalanceUpdate -> m ()
```

Comment:
It seems to be unsafe because `utxoSum` changes non-atomically, but, since the only instance of `MonadTxExtra` is pure state-like type, that is ok.
Easy fix would be to provide modifier instead of setter for `utxoSum`